src/dmd/dsymbol.d

```diff
@@ -2186,41 +2186,61 @@ extern (C++) final class DsymbolTable : RootObject
 {
     AssocArray!(Identifier, Dsymbol) tab;
 
-    // Look up Identifier. Return Dsymbol if found, NULL if not.
+   /***************************
+    * Look up Identifier in symbol table
+    * Params:
+    *   ident = identifer to look up
+    * Returns:
+    *   Dsymbol if found, null if not
+    */
     Dsymbol lookup(const Identifier ident)
     {
         //printf("DsymbolTable::lookup(%s)\n", ident.toChars());
         return tab[ident];
     }
 
-    // Look for Dsymbol in table. If there, return it. If not, insert s and return that.
-    Dsymbol update(Dsymbol s)
+    /**********
+     * Replace existing symbol in symbol table with `s`.
+     * If it's not there, add it.
+     * Params:
+     *   s = replacement symbol with same identifier
+     */
+    void update(Dsymbol s)
     {
-        const ident = s.ident;
-        Dsymbol* ps = tab.getLvalue(ident);
-        *ps = s;
-        return s;
+        *tab.getLvalue(s.ident) = s;
```

Comment:
Revert this, it's not an improvement to the documentation. 

(Also forgot to update dsymbol.h...)

Reply:
This PR is tagged "refactoring".

Updated dsymbol.h.

Reply:
A refactoring should not change the behavior... Have you checked that this change doesn't affect LDC?

Reply:
It doesn't change the behavior - dmd never calls it. As for gdc/ldc, if they're using the return value, it'll give an error, not silently corrupt anything. The return value makes no sense.

Reply:
It occurs to me you and I have different interpretations of what "refactoring" means. To me, it means not changing the behavior of the program. To you (correct me if I'm wrong) it means not changing the API of any functions.

Reply:
Refactorings should retain the behavior  of a component, it's more nuanced than the whole program but doesn't prohibit changes to function signatures.

The problem here is a change to the public API which can break code depending on "DMD as a library" - without providing any real benefit.

But I guess that is acceptable given that the potential breakage isn't silent and can easily be fixed.